applications/FluidDynamicsApplication/custom_utilities/fluid_adjoint_test_utilities.h

```diff
@@ -0,0 +1,111 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#if !defined(KRATOS_FLUID_ADJOINT_TEST_UTILITIES_H_INCLUDED)
+#define KRATOS_FLUID_ADJOINT_TEST_UTILITIES_H_INCLUDED
+
+// System includes
+#include <functional>
+#include <vector>
+
+// External includes
+
+// Project includes
+#include "containers/model.h"
+#include "includes/model_part.h"
+#include "processes/process.h"
+
+// Application includes
+
+namespace Kratos
+{
+///@name Classes
+///@{
+
+class KRATOS_API(FLUID_DYNAMICS_APPLICATION) FluidAdjointTestUtilities
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    using IndexType = std::size_t;
+
+    using NodeType = ModelPart::NodeType;
+
+    ///@}
+    ///@name Static Operations
+    ///@{
+
+    template<class TDataType>
+    static TDataType CalculateRelaxedVariableRate(
+        const double BossakAlpha,
+        const Variable<TDataType>& rVariable,
+        const NodeType& rNode);
+
+    ///@}
+    ///@name Classes
+    ///@{
+
+    template<class TContainerType>
+    class Testing
```

Comment:
Wouldn't it be easier to overload the two methods in this class for elements and conditions and then create a templated method with the implementation? I think that this will ease the usage.

Reply:
done :)